src/effects/VST/VSTEffect.cpp

```diff
@@ -3650,20 +3641,31 @@ void VSTInstanceBase::SizeWindow(int w, int h)
 void VSTInstanceBase::Automate(int index, float value)
 {
    callSetParameterB(index, value);
+
+   // Because we come here when a control on the effect's GUI is moved,
+   // we must update the temporary Effect-owned settings - users might
+   // want to hear what happens when they move a knob.
+   //
+   {
+      // we need a mutex because FetchSettings writes mSettings in the main
+      // thread, but the storing of the settings passed to ::RealtimeProcess
+      // happens in the worker thread
+      auto guard = std::lock_guard{ mSettingsMutex };
+      FetchSettings(mSettings);
```

Comment:
You made these changes in 5.1.  I don't like the mutex.

Does experiment show that this is called on the main thread only in response to slider changes?

I need the answer to that, before I can recommend the correct thing  to do.

Reply:
Please see my comment above https://github.com/audacity/audacity/pull/3278#discussion_r946051875